src/utils/rootkeypackage_utils/src/rootkeypackage_parse.c

```diff
@@ -0,0 +1,824 @@
+
+#include "aduc/rootkeypackage_parse.h"
+#include <aduc/logging.h>
+#include <aduc/string_c_utils.h> // for IsNullOrEmpty
+#include <azure_c_shared_utility/strings.h>
+#include <azure_c_shared_utility/vector.h>
+#include <base64_utils.h> // for Base64URLDecode
+
+EXTERN_C_BEGIN
+
+/**
+ * @brief Frees resources for an ADUC_RootKey object.
+ *
+ * @param node The pointer to the root key object.
+ */
+void ADUC_RootKey_DeInit(ADUC_RootKey* node)
+{
+    if (node == NULL)
+    {
+        return;
+    }
+
+    if (node->kid != NULL)
+    {
+        STRING_delete(node->kid);
+        node->kid = NULL;
+    }
+
+    node->keyType = ADUC_RootKey_KeyType_INVALID;
+
+    if (node->rsaParameters.n != NULL)
+    {
+        CONSTBUFFER_DecRef(node->rsaParameters.n);
+        node->rsaParameters.n = NULL;
+    }
+
+    node->rsaParameters.e = 0;
+}
+
+void ADUC_RootKeyPackage_Hash_DeInit(ADUC_RootKeyPackage_Hash* node)
+{
+    if (node == NULL)
+    {
+        return;
+    }
+
+    node->alg = ADUC_RootKeyShaAlgorithm_INVALID;
+
+    if (node->hash != NULL)
+    {
+        CONSTBUFFER_DecRef(node->hash);
+        node->hash = NULL;
+    }
+}
+
+/**
+ * @brief Parses the version protected property in accordance with rootkeypackage.schema.json
+ *
+ * @param protectedPropertiesObj The root JSON object.
+ * @param[out] outPackage The root key package object to write parsed version data.
+ *
+ * @return The result.
+ */
+ADUC_Result RootKeyPackage_ParseVersion(JSON_Object* protectedPropertiesObj, ADUC_RootKeyPackage* outPackage)
+{
+    ADUC_Result result = { .ResultCode = ADUC_GeneralResult_Failure, .ExtendedResultCode = 0 };
+    unsigned long version = 0;
+
+    if (protectedPropertiesObj == NULL || outPackage == NULL)
+    {
+        result.ExtendedResultCode = ADUC_ERC_UTILITIES_ROOTKEYPKG_UTIL_ERROR_BAD_ARG;
+        goto done;
+    }
+
+    version = json_object_get_number(protectedPropertiesObj, "version");
+    if (version == 0)
```

Comment:
this is also an error condition of json_object_get_number because json_value_get_number will return 0 if json_value_get_type is not JSONNumber.  Will update doc with it.